pycloudlib/oracle/cloud.py

```diff
@@ -0,0 +1,218 @@
+# This file is part of pycloudlib. See LICENSE file for license information.
+"""OCI Cloud type."""
+
+import re
+import oci
+
+from pycloudlib.cloud import BaseCloud
+from pycloudlib.oracle.instance import OciInstance
+from pycloudlib.oracle.utils import wait_till_ready
+from pycloudlib.util import ubuntu_name_to_number
+
+
+class OCI(BaseCloud):
+    """OCI (Oracle) cloud class."""
+
+    _type = 'oci'
+
+    def __init__(self, tag, compartment_id, config_path='~/.oci/config'):
+        """
+        Initialize the connection to OCI.
+
+        OCI must be initialized on the CLI first:
+        https://github.com/cloud-init/qa-scripts/blob/master/doc/launching-oracle.md
+
+        Args:
+            tag: Name of instance
+            compartment_id: A compartment found at
+                https://console.us-phoenix-1.oraclecloud.com/a/identity/compartments
+            config_path: Path of OCI config file
+        """
+        super(OCI, self).__init__(tag)
+        self.compartment_id = compartment_id
+        self.config_path = config_path
+        config = oci.config.from_file(str(config_path))
```

Comment:
Can we do a pre-flight if not os.exists(config_path) here and raise a RuntimeError saying "get your OCI client setup: see <LAUNCHING_ORACLE_URL>?